packages/zoe/src/contracts/newSwap/getCurrentPrice.js

```diff
@@ -0,0 +1,260 @@
+// @ts-check
+
+import '../../../exported';
+
+import { assert, details as X } from '@agoric/assert';
+import { amountMath } from '@agoric/ertp';
+import { multiplyBy, makeRatio, natSafeMath } from '../../contractSupport';
+
+const { ceilDivide } = natSafeMath;
+const BASIS_POINTS = 10000;
+
+/**
+ * Build functions to calculate prices for multipoolAutoswap. Four methods are
+ * returned, as two complementary pairs. In one pair of methods the caller
+ * specifies the amount they will pay, and in the other they specify the amount
+ * they wish to receive. The two with shorter names (getOutputForGivenInput,
+ * getInputForGivenOutput) are consistent with the uniswap interface. They each
+ * return a single amount. The other two return { amountIn, centralAmount,
+ * amountOut }, which specifies the best exchange consistent with the request.
+ * centralAmount is omitted from these methods' results in the publicFacet.
+ *
+ * @param {(brand: Brand) => boolean} isSecondary
+ * @param {(brand: Brand) => boolean} isCentral
+ * @param {(brand: Brand) => Pool} getPool
+ * @param {Brand} centralBrand
+ * @param {bigint} poolFeeBP - fee (BP) to the pool in every swap.
+ * @param {bigint} protocolFeeBP - fee (BP) charged in RUN to every swap.
+ */
+
+export const makeGetCurrentPrice = (
+  isSecondary,
+  isCentral,
+  getPool,
+  centralBrand,
+  poolFeeBP,
+  protocolFeeBP,
+) => {
+  const protocolFeeRatio = makeRatio(protocolFeeBP, centralBrand, BASIS_POINTS);
+  const halfPoolFeeBP = ceilDivide(poolFeeBP, 2);
```

Comment:
I think it's simpler if we don't try to handle secondary to secondary as a special case, and we just charge double the fees. That's expected.

Reply:
My intent was not to split the fee, but rather extract the protocol fee from whatever fees we were charging. That's consistent with virtual pools or the existing switch statement approach. I need to read this code more to provide any more useful input.